kion/internal/kionclient/helper.go

```diff
@@ -130,21 +130,24 @@ func FlattenGenericIDArray(d *schema.ResourceData, key string) []int {
 func FlattenGenericIDPointer(d *schema.ResourceData, key string) *[]int {
 	uid := d.Get(key)
 
-	// test for set
-	_, isSet := uid.(*schema.Set)
-	if isSet {
-		uid = uid.(*schema.Set).List()
-	}
-
-	uids := make([]int, 0)
-	for _, item := range uid.([]interface{}) {
-		v, ok := item.(map[string]interface{})
-		if ok {
-			uids = append(uids, v["id"].(int))
+	switch v := uid.(type) {
+	case []interface{}:
+		uids := make([]int, len(v))
+		for i, item := range v {
+			uids[i] = item.(int)
+		}
+		return &uids
+	case *schema.Set:
+		setList := v.List()
+		uids := make([]int, len(setList))
+		for i, item := range setList {
+			m := item.(map[string]interface{})
+			uids[i] = m["id"].(int)
 		}
+		return &uids
+	default:
+		return nil
```

Comment:
What does this change do and why was it required as part of this PR?

In the `[]interface{}` case, is an array of int ids normally passed in?

Reply:
The changes to the `FlattenGenericIDPointer` function were to handle different data types (`[]interface{} and `*schema.Set`) that I ran across with Terraform in how it stored schema attributes for users and groups.  The way we do project enforcements for users and groups is that we have a whole separate endpoint for adding users/groups, and I discovered the way the schema attributes were being stored was different than anywhere else that I've run into.  

In the context of the `AddProjectEnforcementUsers` and `RemoveProjectEnforcementUsers` functions:

The function now reliably converts user and user group IDs into a `[]int` format.

The `[]interface{}` case is common when the schema attribute is defined as a list, and the `*schema.Set` case handles scenarios where the attribute is defined as a set. This handles both cases.  For example, `user_group_ids` and `user_ids` are stored as `TypeList` [here](https://github.com/kionsoftware/terraform-provider-kion/pull/75/files#diff-d85aab7d9780cf41a75b8db8cb6469175d7b434752e2598daaa163e604efc974R104), whereas [here](https://github.com/kionsoftware/terraform-provider-kion/blob/main/kion/resource_project.go#L83) they're stored as `TypeSet`